Library/Homebrew/brew.sh

```diff
@@ -469,6 +469,15 @@ Your Git executable: $(unset git && type -p ${HOMEBREW_GIT})"
 
   HOMEBREW_LINUX_MINIMUM_GLIBC_VERSION="2.13"
   unset HOMEBREW_MACOS_SYSTEM_RUBY_NEW_ENOUGH
+
+  HOMEBREW_CORE_REPOSITORY_ORIGIN="$("${HOMEBREW_GIT}" -C "${HOMEBREW_CORE_REPOSITORY}" remote get-url origin)"
+  if [[ "${HOMEBREW_CORE_REPOSITORY_ORIGIN}" = "https://github.com/Homebrew/homebrew-core" ]]
+  then
+    # If the remote origin has been set to Homebrew/homebrew-core by the install script,
+    # then we are in the case of a new installation of brew, using Homebrew/homebrew-core as a Linux core repository.
+    # In that case, set HOMEBREW_FORCE_HOMEBREW_ON_LINUX=1 to set the right HOMEBREW_BOTTLE_DOMAIN below.
+    export HOMEBREW_FORCE_HOMEBREW_ON_LINUX="1"
```

Comment:
This does a few things I'm not sure is desirable to set it here. Instead, I think we should fix/remove the existing checks to stop complaining if a new variable is set e.g. `HOMEBREW_LINUX_HOMEBREW_CORE_MIGRATION="1"` or similar until we've migrated everyone.

Reply:
Done, I introduced a new env variable just for the migration and added a TODO comment that we will need to clean this up later on.

Once we have decommissioned linuxbrew-core we will need to make some cleanups in brew and in test-bot to remove any reference to linuxbrew-core anyway.